kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java

```diff
@@ -9211,7 +9211,11 @@ protected Map fixTypeAndId(String id, String type)
 		extType.put("id", id);
 		if (type == null) type = "";
 		extType.put("type", type);
-		String extension = Validator.getFileExtension(id);
+		
+		String[] idSegments;
+		idSegments = id.split("/");
+		String filename = idSegments.length > 0 ? idSegments[idSegments.length - 1]:id;
+		String extension = Validator.getFileExtension(filename);
```

Comment:
No, it's not a system path and we don't ever want to split on a Windows separator. It might be the case that you can't ever pass a '\' into this method but I didn't debug that far.